swagger.yaml

```diff
@@ -0,0 +1,221 @@
+  version: "1.0.0"
+  title: "Filter API"
+  license:
+    name: "Open Government Licence v3.0"
+    url: "http://www.nationalarchives.gov.uk/doc/open-government-licence/version/3/"
+basePath: "/v1"
+tags:
+- name: "Jobs"
+  description: "Used for storing the state of a customised dataset and retrieving the results"
+schemes:
+- "http"
+parameters:
+  jobId:
+    name: jobId
+    type: string
+    required: true
+    description: "The unique jobId for customising a dataset"
+    in: path
+  dimension:
+    name: dimension
+    schema:
+      $ref: '#/definitions/Dimension'
+    required: true
+    description: "A dimension to filter the dataset"
+    in: body
+  state:
+    name: state
+    schema:
+      $ref: '#/definitions/State'
+    required: true
+    description: "The requested state of the job"
+    in: body
+  newJob:
+    name: newJob
+    schema:
+      $ref: '#/definitions/NewJob'
+    required: true
+    description: "A new job to create"
+    in: body
+paths:
+  /job:
+    post:
+      tags:
+      - "Jobs"
+      summary: "Create a filter job for a dataset"
+      description: "Create a job so that dimensions can be added to filter a dataset"
+      produces:
+      - "application/json"
+      parameters:
+      - $ref: '#/parameters/newJob'
+      responses:
+        200:
+          description: "Job was created"
+          schema:
+            $ref: '#/definitions/JobInstance'
+        500:
+          description: "Failed to process the request due to an internal error"
+  /jobs/{jobId}:
+    get:
+      tags:
+      - "Jobs"
+      summary: "Get a description of a created job"
+      description: "A jobId is required to access a customised dataset job. To create a jobId see the `Dataset API`"
+      produces:
+      - "application/json"
+      parameters:
+      - $ref: '#/parameters/jobId'
+      responses:
+        200:
+          description: "The job was found and a `JobState` is returned"
+          schema:
+            $ref: '#/definitions/JobState'
+        404:
+          description: "Job not found"
+        500:
+          description: "Failed to process the request due to an internal error"
+  /jobs/{jobId}/filter:
+    put:
+      tags:
+      - "Jobs"
+      summary: "Add a dimension to filter the dataset"
+      description: "The dimension can only be added into the job if the state is still set to `created` otherwise 400 status code is returned"
+      produces:
+      - "application/json"
+      parameters:
+      - $ref: '#/parameters/jobId'
+      - $ref: '#/parameters/dimension'
+      responses:
+        200:
+          description: "Filter list was updated"
+        400:
+          description: "The job has been locked as it has been `submitted` to be processed"
+        404:
+          description: "Job not found"
+        500:
+          description: "Failed to process the request due to an internal error"
+  /jobs/{jobId}/state:
```

Comment:
As state is a property of the job, it should be a PUT to /jobs/{jobId} specifying only the state in the body.